alex-go-nguyen/blogsite
# Feature: Build commons feature and setup deploy

# Issue
I build some feature below
- Homepage
- Article detail page
- Comment
- Category page
- Login / Logout
- Registration
- Personal feature (Update profile, password)
- Author detail page
- Post article
- Deploy
## Summary
- Homepage
  - Show newest posts
- Article detail page
  - Show content of article
- Comment
  - User can put their comment to article
- Category page
  - Show article filter by category
- Login / Logout
  - User login using their account
- Registration
  - User can register new account
- Personal feature (Update profile, password)
  - User logged in can modified their information
- Author detail page
  - Show profile of author and their article
- Post article
  - User can write new article and post it
- Deploy
  - Deploy website to vercel
### Media showcase

https://github.com/alex-go-nguyen/blogsite/assets/133078261/a2fca349-114a-47f6-ba32-1cfd4ffe6bfe

- Home page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/2babd772-f270-44ee-a4e3-63767c197d8b)
- Article detail page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/291377d2-eb03-4a46-9dff-bbbcebe0b717)
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/f655bf3c-98f1-419f-993d-1c81444c4450)
- Category page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/881920d8-69f3-46eb-8301-91d82c591a86)
- Register page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/fd4b5b40-ffab-4c99-a9a6-d9891459da22)
- Login page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/befffd1e-3da3-4210-8db1-bb88e0a545af)
- Account page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/2744fbbe-5a4f-4adc-bbed-45ad129074de)
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/6df37a10-1ed9-4e5c-adfc-f23744d0b85c)
- Post article
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/3b23fc15-77ad-42e1-a401-757dbc6abdf0)
- Author page
![image](https://github.com/alex-go-nguyen/blogsite/assets/133078261/812624be-4868-49e3-8ad6-117dfa4fc72e)

### Testing instructions:

- For the feature, you can access to my deployment website ``blogowl.vercel.app`` to check it
- For the test components, you can try follow in Readme.md

## Reviews

### Author
-   [x] I have reviewed my own code (all files marked as viewed in GitHub).
-   [x] I have tested (via the postman) each of the interfaces/features related to the files changed/added.
